packages/ld/ld.ts

```diff
@@ -8,62 +8,71 @@ export type LdOptions<T extends object> = {
   onError: (err: unknown) => void;
 } & RequestInit;
 
-const table = new Map()
+type TableMap<T> = Map<string, T|undefined>;
 
-function proxyfy<T extends object>(obj: T, options: LdOptions<T>): T {
+const cachedTable: TableMap<unknown> = new Map()
+
+function proxify<T extends object>(obj: T, options: LdOptions<T>): T {
   if (undefined === options.root) {
-    table.clear()
+    cachedTable.clear()
     options.root = obj
   }
 
   return new Proxy<T>(obj, {
-    get: (target, prop, receiver) => {
-      const value = Reflect.get(target, prop, receiver);
+    get: (target: T, prop: string, receiver: unknown) => {
+      const value: string extends keyof T ? T[string] : unknown = Reflect.get(target, prop, receiver);
 
       if (typeof value === 'object' && value !== null) {
-        return proxyfy<any>(value, options);
+        return proxify<any>(value, options);
```

Comment:
do you have any idea of we could use `typeof value` instead of `any`?

Reply:
There's a few problems here that I couldn't (yet) fix:
* `value` is presumably no longer an instance of `T`. It's not even an object anymore.
* Since `options` is `LdOptions<T>`, with the `root: T` property and type, we should instead add 2 different types I guess, or truly enforce `options.root` to be defined at first, even if we have to define it twice when proxying objects.
* This could also be enforced somehow by making a difference between `LdOptions<T>` which is only for `ld`, and a new `ProxyOptions` type that would require a specific root and type to be defined beforehand (and possibly immutable at call-time)

Reply:
Indeed, hard problem. Not sure there's any benefit in duplicating options, nor to have 2 functions doing almost the same job. I don't think depth is important (as you choose what links we fetch) and the code already handles circular references.

Reply:
I think it might be better to keep it as `any`, for now at least. Typescript will complain, but well, when doesn't it? 😅